### ISSUE TITLE
Add timeline REST API

### ISSUE THREAD

**Issue by mikkonie** [bihealth/sodar-core#1350]
Recently we have discussed use cases where this could be useful, for e.g. detecting when certain data has been edited in a project.

The API should be read-only. There should be no need for an external system to create or modify timeline events, apps do that when actions are performed.

This should be done after #1278, so we can follow the new standards and not have to redo our work.

**TBD:** Target v1.0 or v1.1? I'm placing this under v1.0 for now, but if that milestone starts to feel too huge I might postpone. This is not exactly urgent at this point.

## Tasks

- [x] Add serializers
- [x] Add views
- [x] Add view tests
- [x] Add permission tests
- [x] Update docs
- [x] Add displaying/hiding extra data based on perms

**Comment by mikkonie** [comment]
We've identified new use cases for this regarding data stewardship, so this should definitely go in v1.0.

**Comment by mikkonie** [comment]
Done.
